unikraft/app/kernel.go

```diff
@@ -0,0 +1,388 @@
+// SPDX-License-Identifier: BSD-3-Clause
+// Copyright (c) 2023, Unikraft GmbH and The KraftKit Authors.
+// Licensed under the BSD-3-Clause License (the "License").
+// You may not use this file except in compliance with the License.
+package app
+
+import (
+	"bytes"
+	"compress/gzip"
+	"context"
+	"debug/elf"
+	"encoding/binary"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"kraftkit.sh/kconfig"
+	makefile "kraftkit.sh/make"
+	"kraftkit.sh/unikraft/core"
+	"kraftkit.sh/unikraft/lib"
+)
+
+var uk_ibinfo_section_name = ".uk_libinfo"
+
+// This type contains informations about how the components of a unikernel were built.
+// Those can be either the core(unikraft) or the libraries, the difference being that
+// in the case of the core, the name of the component is empty.
+type componentInfoRecord struct {
```

Comment:
Let's publically expose this struct and all the attributes. Please also add `json` and `yaml` attribute tags.